src/test/java/com/microsoft/azure/relay/SendReceiveTest.java

```diff
@@ -204,6 +204,24 @@ public void httpLargePOSTAndLargeResponseTest() throws IOException {
         httpRequestSender("POST", largeStr, largeStr);
 	}
 	
+	@Test
+	public void httpWriteSmallThenSmallResponseTest() throws IOException {
```

Comment:
>httpWriteSmallThenSmallResponseTest [](start = 13, length = 35)

Does anything in the test suite fail without the send async lock that's being introduced?

Reply:
Runs fine for now with synchronized writes, will do async writes in separate PR

---
In reply to: [292063905](https://github.com/Azure/azure-relay-java/pull/45#discussion_r292063905) [](ancestors = 292063905)